packages/react-aria-components/docs/Tree.mdx

```diff
@@ -753,6 +766,1059 @@ Use the `renderEmptyState` prop to customize what the `Tree` will display if the
 
 </details>
 
+## Drag and drop
+
+Tree supports drag and drop interactions when the `dragAndDropHooks` prop is provided using the <TypeLink links={docs.links} type={docs.exports.useDragAndDrop} /> hook. Users can drop data on the tree as a whole, on individual items, insert new items between existing ones, or move items within the tree hierarchy.
+
+React Aria supports traditional mouse and touch based drag and drop, but also implements keyboard and screen reader friendly interactions. Users can press <Keyboard>Enter</Keyboard> on a draggable element to enter drag and drop mode. Then, they can press <Keyboard>Tab</Keyboard> to navigate between drop targets. A droppable collection is treated as a single drop target, so that users can easily tab past it to get to the next drop target. Within a droppable tree, keys such as <Keyboard>ArrowDown</Keyboard> and <Keyboard>ArrowUp</Keyboard> can be used to select a _drop position_, such as on an item, or between items.
+
+Items with children can be expanded or collapsed during keyboard drag and drop mode by using the <Keyboard>ArrowRight</Keyboard> and <Keyboard>ArrowLeft</Keyboard> keys, or the <Keyboard>Option</Keyboard> (<Keyboard>Alt</Keyboard> on Windows) + <Keyboard>Enter</Keyboard> keys. Screen reader users can expand and collapse items with children by navigating to and activating a visually hidden button that gets announced.
+
+Draggable tree items must include a focusable drag handle using a `<Button slot="drag">`. This enables keyboard and screen reader users to initiate drag and drop. The `MyTreeItemContent` component defined in the [reusable wrappers](#reusable-wrappers) section above can be extended to include this automatically when the tree allows dragging.
+
+See the [drag and drop introduction](dnd.html) to learn more.
+
+### Reorderable
```

Comment:
This example feels pretty confusing, mainly because there aren't that many items that can be reordered (each level has only 1 or 2 items in it). Wondering if we should show onMove example first since that will probably be more common?